File: core/src/com/alexpi/awesometanks/widget/GameButton.java
package com.alexpi.awesometanks.widget;

import com.alexpi.awesometanks.utils.Settings;
import com.alexpi.awesometanks.utils.Styles;
import com.badlogic.gdx.assets.AssetManager;
import com.badlogic.gdx.audio.Sound;
import com.badlogic.gdx.scenes.scene2d.InputEvent;
import com.badlogic.gdx.scenes.scene2d.ui.TextButton;
import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;

public class GameButton extends TextButton {
    public interface OnClickListener{
        void onClick();
    }

    public GameButton(AssetManager assetManager, final OnClickListener onClickListener, String text){
        super(text, Styles.getTextButtonStyle1(assetManager));
        final Sound clickSoundDown = assetManager.get("sounds/click_down.ogg",Sound.class);
        final Sound clickSoundUp = assetManager.get("sounds/click_down.ogg",Sound.class);

        addListener(new ClickListener(){
            @Override
            public boolean touchDown(InputEvent event, float x, float y, int pointer, int button) {
                if(Settings.INSTANCE.getSoundsOn()) clickSoundDown.play();
                return super.touchDown(event, x, y, pointer, button);
            }

            @Override
            public void clicked(InputEvent event, float x, float y) {
                if(Settings.INSTANCE.getSoundsOn()) clickSoundUp.play();
                if(onClickListener != null) onClickListener.onClick();
            }
        });
    }
}

File: core/src/com/alexpi/awesometanks/screens/widget/GameButton.kt
package com.alexpi.awesometanks.screens.widget

import com.alexpi.awesometanks.game.module.Settings.soundsOn
import com.badlogic.gdx.assets.AssetManager
import com.badlogic.gdx.audio.Sound
import com.badlogic.gdx.scenes.scene2d.InputEvent
import com.badlogic.gdx.scenes.scene2d.ui.TextButton
import com.badlogic.gdx.scenes.scene2d.utils.ClickListener

class GameButton(assetManager: AssetManager, text: String = "", var onClickListener: (() -> Unit)? = null) :
    TextButton(text, Styles.getTextButtonStyle1(assetManager)) {

    init {
        val clickSoundDown = assetManager.get("sounds/click_down.ogg", Sound::class.java)
        val clickSoundUp = assetManager.get("sounds/click_down.ogg", Sound::class.java)


        addListener(object : ClickListener() {
            override fun touchDown(
                event: InputEvent,
                x: Float,
                y: Float,
                pointer: Int,
                button: Int
            ): Boolean {
                if (soundsOn) clickSoundDown.play()
                return super.touchDown(event, x, y, pointer, button)
            }

            override fun clicked(event: InputEvent, x: Float, y: Float) {
                if (soundsOn) clickSoundUp.play()
                onClickListener?.invoke()
            }
        })
    }
}
